spec/decorators/artist_decorator_spec.rb

```diff
@@ -0,0 +1,14 @@
+require 'rails_helper'
+
+describe 'Artist Decorator' do
+  describe 'decoration associations' do
+    it 'decorates associated songs' do
+      create(:song)
+      artists = Artist.includes(:songs).decorate
+      song = artists.first.songs.first
+
+      expect(song.respond_to?('full_title')).to eq true
```

Comment:
Am a bit confused as to why the song decoration is being tested in the `ArtistDecorator`?

Is it trying to test

```
+class ArtistDecorator < Draper::Decorator
+  delegate_all
+  decorates_association :songs
+end
```

`+  decorates_association :songs` ?

If that's the case, I'd do something like:

```
artist = create :artist, name: 'Pixies'
create :song, artist: artist, duration: 60, album: 'Doolittle', title: 'Wave of Mutilation'

decorated_artist = artist.decorate
song = decorated_artist.songs.first

expect(song.duration_in_minutes_and_seconds).to eq '1:00'
expect(song. full_title).to eq 'Pixies - Doolittle - Wave of Mutilation'
```

and in the factory for artist, use the trait to already create a